sdk/eventhub/Azure.Messaging.EventHubs/src/EventProcessorClient.cs

```diff
@@ -287,194 +437,404 @@ public class EventProcessorClient : EventProcessorBase<PartitionContext>, IAsync
+                            }
+
+                            // We expect the token source to be null, but we are playing safe.
+
+                            RunningTaskTokenSource?.Cancel();
+                            RunningTaskTokenSource = new CancellationTokenSource();
+
+                            // Start the main running task.  It is responsible for managing the active partition processing tasks and
+                            // for partition load balancing among multiple event processor instances.
+
+                            ActiveLoadBalancingTask = RunAsync(RunningTaskTokenSource.Token);
+                        }
+                    }
                 }
-                else if (ownership.SequenceNumber.HasValue)
+                finally
                 {
-                    startingPosition = EventPosition.FromSequenceNumber(ownership.SequenceNumber.Value);
+                    RunningTaskSemaphore.Release();
                 }
-
-                // TODO: it might be troublesome to let the users add running tasks by themselves.  If the user adds a custom
-                // processing task that's not RunPartitionProcessingAsync, how would the base stop it?  It would not have a cancellation
-                // token to do so.
-
-                ActivePartitionProcessors[context.PartitionId] = RunPartitionProcessingAsync(context.PartitionId, startingPosition, Options.MaximumReceiveWaitTime, Options.RetryOptions, Options.TrackLastEnqueuedEventProperties);
-            }
-            catch (Exception)
-            {
-                // If processing task creation fails, we'll try again on the next time this method is called.  This should happen
-                // on the next load balancing loop as long as this instance still owns the partition.
-                // TODO: delegate the exception handling to an Exception Callback.  Do we really need a try-catch here?
             }
         }
 
         /// <summary>
-        ///   The handler to be called once event processing stops for a given partition.
+        ///   Starts the event processor.  In case it's already running, nothing happens.
+        /// </summary>
+        ///
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> instance to signal the request to cancel the operation.</param>
+        ///
+        /// <exception cref="EventHubsClientClosedException">Occurs when this <see cref="EventProcessorClient" /> instance is already closed.</exception>
+        /// <exception cref="InvalidOperationException">Occurs when this method is invoked without <see cref="ProcessEventAsync" /> or <see cref="ProcessErrorAsync" /> set.</exception>
+        ///
+        public virtual void StartProcessing(CancellationToken cancellationToken = default) => StartProcessingAsync(cancellationToken).GetAwaiter().GetResult();
+
+        /// <summary>
+        ///   Stops the event processor.  In case it isn't running, nothing happens.
```

Comment:
Same line of thinking as with `Start` with respect to the summary and token description.